tests/testthat/test-dup.R

```diff
@@ -13,3 +13,20 @@ test_that("0 duplicates equals empty string", {
 test_that("uses tidyverse recycling rules", {
   expect_error(str_dup(1:2, 1:3), class = "vctrs_error_incompatible_size")
 })
+
+test_that("uses sep argument", {
+  expect_equal(str_dup("a", 3, sep = ";"), "a;a;a")
+  expect_equal(str_dup("abc", 2, sep = "-"), "abc-abc")
+  expect_equal(str_dup("a", 3, sep = ";"), "a;a;a")
+  expect_equal(str_dup("abc", 2, sep = "-"), "abc-abc")
```

Comment:
I think you could drop these tests because they're all basically executing all the same code.

Reply:
done in subsequent commit